cypress/e2e/flow-builder/blocks/moving.cy.ts

```diff
@@ -0,0 +1,74 @@
+import {IBlock} from '@floip/flow-runner'
+import Chainable = Cypress.Chainable
+
+describe('Moving a block', () => {
+  beforeEach(() => {
+    cy.createFlow({
+      label: 'Moving a block',
+    })
+
+    cy.addBlock(['Content', 'Message']).then(blockId => {
+      cy.get(`[data-cy="block--${blockId}"]`)
+        .find('[data-cy^="block-handle--"]')
+        .as('blockArea')
+    })
+
+    // wait for flow validation to succeed and flow validation toolbar to disappear
+    // to avoid interference with block coordinates
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(200)
+  })
+
+  it('moves the block', () => {
+    const deltaX = 200
+    const deltaY = 300
+
+    // eslint-disable-next-line arrow-body-style
+    return getBlockVisualCoordinates().then(({x: oldVisualX, y: oldVisualY}) => {
+      return getBlockMetadataCoordinates().then(({x: oldMetadataX, y: oldMetadataY}) => {
+        cy.get('@blockArea').move({deltaX, deltaY})
+
+        // eslint-disable-next-line arrow-body-style
+        return getBlockVisualCoordinates().then(({x: newVisualX, y: newVisualY}) => {
+          return getBlockMetadataCoordinates().then(({x: newMetadataX, y: newMetadataY}) => {
+            cy.log('old visual x,y:', oldVisualX, oldVisualY)
+            cy.log('new visual x,y:', newVisualX, newVisualY)
+
+            cy.wrap(newVisualX - oldVisualX).should('eq', deltaX)
```

Comment:
any particular idea why do you propose to assert the visual coordinates & the metadata coordinates? No major concern, just curious.

Reply:
At first, I was trying to do the visual testing - checking the `left` and `top` of the block - but it didn't work.
So I switched to Vuex instead. It worked.
And then I realized that I was doing something wrong with visual testing and fixed it.
So I have ended up with two working approaches, and I just kept them both.